### PR TITLE
add waitFor msg

### PR DESCRIPTION
**Pull request by snewcomer**
Opens up `waitFor` to have a `timeoutMessage`.

### CODE REVIEW

File: addon-test-support/@ember/test-helpers/dom/wait-for.js [hunk 1]
```diff
@@ -15,7 +15,7 @@ import { nextTickPromise } from '../-utils';
   @param {number} [options.count=null] the number of elements that should match the provided selector (null means one or more)
   @returns {Element|Array<Element>} the element (or array of elements) that were being waited upon
 */
-export default function waitFor(selector, { timeout = 1000, count = null } = {}) {
+export default function waitFor(selector, { timeout = 1000, count = null, timeoutMessage = 'waitFor timed out' } = {}) {
```
Comment (snewcomer) [inline addon-test-support/@ember/test-helpers/dom/wait-for.js]: I can revert the default and keep `waitUntil timed out` if you would like.

File: tests/unit/dom/wait-for-test.js [hunk 2]
```diff
@@ -71,11 +71,11 @@ module('DOM Helper: waitFor', function(hooks) {
 
     let start = Date.now();
     try {
-      await waitFor('.something', { timeout: 100 });
+      await waitFor('.something', { timeout: 100, timeoutMessage: '.something timed out' });
```
Comment (rwjblue) [inline tests/unit/dom/wait-for-test.js]: Lets leave this test alone, and add another that is specifically for modifying the timeout message...

File: addon-test-support/@ember/test-helpers/dom/wait-for.js (hunk 1 above)
Comment (rwjblue) [reply addon-test-support/@ember/test-helpers/dom/wait-for.js]: Nah, I think this is fine...

**Comment by rwjblue** [comment]
@snewcomer - 👋 looks like some linting failures here still but if you fix those and rebase against master, CI should be green again...

**Comment by rwjblue** [comment]
Thank you!

**Pull request merged**